boards/PDM/Inc/Io/Io_Efuse.h

```diff
@@ -1,164 +1,53 @@
+    EFUSE_CHANNEL_FAN,
+    EFUSE_CHANNEL_DI_LHS,
+    EFUSE_CHANNEL_DI_RHS,
+    NUM_EFUSE_CHANNELS,
+} EfuseChannel;
 
 /**
- * Get the channel 0 current reading in Amps [A] for the given efuse.
- * @param efuse Pointer to the given efuse
- * @return The channel 0 current if the current was measured, else NAN
+ * Enable or disable the provided efuse channel.
+ * @param channel Channel to enable/disable
+ * @param enabled Enable if enabled is true, disable if false
  */
-float Io_Efuse_GetChannel0Current(struct Efuse_Context *const efuse);
+void Io_Efuse_SetChannel(EfuseChannel channel, bool enabled);
 
 /**
- * Get the channel 1 current reading in Amps [A] for the given efuse.
- * @param efuse Pointer to the given efuse
- * @return The channel 1 current if the current was measured, else NAN
+ * Get whether or not an efuse channel is enabled.
+ * @return If efuse channel is enabled
  */
-float Io_Efuse_GetChannel1Current(struct Efuse_Context *const efuse);
+bool Io_Efuse_IsChannelEnabled(EfuseChannel channel);
 
 /**
- * Write data to a specific serial input register on the given efuse.
- * @param register_address Serial input register being written to
- * @param register_value The value being written to the serial input register
- * @param efuse Pointer to efuse structure being written to
- * @return EXIT_CODE_OK if the write was successful
- *         EXIT_CODE_TIMEOUT if the SPI write timed-out
+ * Get the current drawn by an efuse channel.
+ * @param channel Channel to read current from
+ * @return The current read from the provided channel, in A
  */
-ExitCode Io_Efuse_WriteRegister(uint8_t register_address, uint16_t register_value, struct Efuse_Context *const efuse);
+float Io_Efuse_GetChannelCurrent(EfuseChannel channel);
 
 /**
- * Read data from a specific serial output register for the given efuse.
- * @param register_address Serial output register being read from
- * @param register_value The value read back from the serial output register
- * @param efuse Pointer to efuse structure being read from
- * @return EXIT_CODE_OK if the read was successful
- *         EXIT_CODE_TIMEOUT if the SPI read timed-out
+ * De-latch faults on a given efuse channel. The VND5T100LAx-E will latch itself
+ * off in the case of an overtemperature or overcurrent condition. To clear these faults,
+ * a low pulse must be sent on the standby reset line.
+ *
+ * Note that each efuse IC has 2 channels, and delatching will affect them both. We have 4 efuse ICS, with channels:
+ * 1. Efuse 1: AIR and LVPWR
+ * 2. Efuse 2: EMETER and AUX
+ * 3. Efuse 3: DRS and FAN
+ * 4. Efuse 4: DI_LHS and DI_RHS
+ *
+ * So be aware that, for example, a delatching faults on the AIR channel will also delatch faults on the LVPWR line.
+ *
+ * @param channel Channel to delatch faults for
  */
-ExitCode Io_Efuse_ReadRegister(uint8_t register_address, uint16_t *register_value, struct Efuse_Context *const efuse);
+void Io_Efuse_DelatchChannelFaults(EfuseChannel channel);
```

Comment:
In my opinion, this should be renamed to Io_Efuse_StandbyReset(EfuseChannel channel);
DelatchChannelFaults just sounds like it returns which efuse faulted.

Reply:
good call will change